go-controller/pkg/factory/factory_test.go

```diff
@@ -571,16 +591,19 @@ var _ = Describe("Watch Factory Operations", func() {
 		Eventually(c.getUpdated, 2).Should(Equal(10))
 		Eventually(c.getDeleted, 2).Should(Equal(10))
 		for _, ot := range testNodes {
+			ot.mu.Lock()
 			Expect(ot.added).Should(Equal(2))
```

Comment:
the async assert is not done in the 3 lines before the loop?
```
		Eventually(c.getAdded, 2).Should(Equal(10))
		Eventually(c.getUpdated, 2).Should(Equal(10))
		Eventually(c.getDeleted, 2).Should(Equal(10))
```

Reply:
Huh. The Gomega authors apparently don't understand what "asynchronous" means. I assumed that "asynchronous assertions" ran asynchronously, but in fact, `Eventually()` just blocks until it times out or the condition passes. Nice.

Reply:
(Well, ok... it's not "an asynchronous assertion that something happens", it's "a (synchronous) assertion that something happens asynchronously". :man_shrugging: )

Reply:
Eventually() was the way to know when our end state was reached in the tests. We try to do something, and then wait a while and eventually (hah!) that thing would be true, or the test would fail. Which would be the same general scheme as if we just waited for the queues to (eventually!) be drained...

Reply:
except that this way requires us to know for sure how long it will take for the queues to drain in the worst case, and Antonio already had to update the timeouts somewhere (maybe in a different PR?) because they were too short in some cases

Reply:
I think that you refer to this PR https://github.com/ovn-org/ovn-kubernetes/pull/1498 , that was the problem, there were no polling so if the queue was't drained before the assert the test failed